server/src/main/scala/viper/gobraserver/GobraServer.scala

```diff
@@ -51,50 +53,61 @@ object GobraServer extends GobraFrontend {
     _server.start()
     VerifierState.flushCachedDiagnostics()
   }
-  
 
-  private def serverExceptionHandling(fileData: FileData, resultFuture: Future[VerifierResult]) {
+  def restart() {
+    _server.stop()
+    ViperBackends.ViperServerBackend.resetServer()
+    init(_options)
+    start()
+  }
+
+  private def serverExceptionHandling(fileData: FileData, resultFuture: Future[VerifierResult]): Future[VerifierResult] = {
 
     val fileUri = fileData.fileUri
-    val filePath = fileData.filePath
 
-    resultFuture.onComplete {
-      case Success(_) => // ignore -> handled by reporter
+    // do some post processing if verification has failed
+    resultFuture.recoverWith({ case exception =>
+      // restart Gobra Server and then update client state
+      restart()
 
-      case Failure(exception) =>
+      exception match {
+        case e: Violation.LogicException =>
+          VerifierState.removeDiagnostics(fileUri)
+          val overallResult = Helper.getOverallVerificationResultFromException(fileUri, e)
 
-        exception match {
-          case e: Violation.LogicException =>
-            VerifierState.removeDiagnostics(fileUri)
-            val overallResult = Helper.getOverallVerificationResultFromException(fileUri, e)
+          VerifierState.updateVerificationInformation(fileUri, Right(overallResult))
 
-            VerifierState.updateVerificationInformation(fileUri, Right(overallResult))
 
+          if (fileUri == VerifierState.openFileUri) {
+            VerifierState.publishDiagnostics(fileUri)
+          }
```

Comment:
If the error was caused by a different file, are they still shown in the error/problem view of VS Code?

Reply:
Do you mean in the context of packages? In any case this is a good point that should go into the readme that we do not support yet packages (I need to try what happens if you try to do so)

Reply:
I've created issue #13 for adding support of packages. At the moment, Gobra IDE explicitly sets `inputFiles` to the currently selected file in VSCode (the check for other files belonging to the same package is skipped as this only happens when using `ScallopGobraConfig`. `includeDirs` is left at its default value being empty